src/Arcus.Messaging.Pumps.Abstractions/Resiliency/IMessagePumpCircuitBreaker.cs

```diff
@@ -23,12 +23,83 @@ public interface IMessagePumpCircuitBreaker
         /// <param name="configureOptions">The optional user-configurable options to manipulate the workings of the message pump interaction.</param>
         /// <exception cref="ArgumentException">Thrown when the <paramref name="jobId"/> is blank.</exception>
         Task PauseMessageProcessingAsync(string jobId, Action<MessagePumpCircuitBreakerOptions> configureOptions);
+    }
+
+    /// <summary>
+    /// Represents the available states in the <see cref="MessagePumpCircuitState"/> in which the message pump can transition into.
+    /// </summary>
+    internal enum CircuitBreakerState
+    {
+        Closed, HalfOpen, Open
+    }
+
+    /// <summary>
+    /// Represents the available states in which the <see cref="MessagePump"/> is presently in within the circuit breaker context
+    /// </summary>
+    public sealed class MessagePumpCircuitState
+    {
+        private readonly CircuitBreakerState _state;
+
+        private MessagePumpCircuitState(CircuitBreakerState state) : this(state, new MessagePumpCircuitBreakerOptions())
+        {
+        }
+
+        private MessagePumpCircuitState(CircuitBreakerState state, MessagePumpCircuitBreakerOptions options)
+        {
+            _state = state;
+
+            Options = options ?? throw new ArgumentNullException(nameof(options));
+        }
 
         /// <summary>
-        /// Continue the process of receiving messages in the message pump after a successful message handling.
+        /// Gets the boolean flag that indicates whether the message pump circuit breaker state is in the 'Closed' state,
+        /// and should start retrieving messages.
         /// </summary>
-        /// <param name="jobId">The unique identifier to distinguish the message pump in the application services.</param>
-        /// <exception cref="ArgumentException">Thrown when the <paramref name="jobId"/> is blank.</exception>
-        Task ResumeMessageProcessingAsync(string jobId);
+        public bool IsClosed => _state is CircuitBreakerState.Closed;
+
+        /// <summary>
+        /// Gets the boolean flag that indicates whether the message pump circuit breaker state is in the 'Open' state,
+        /// and should stop retrieving messages.
+        /// </summary>
+        public bool IsOpen => _state is CircuitBreakerState.Open;
```

Comment:
I'm wondering why we don't expose the `CircuitBreakerState` itself (the value of the `_state` member) instead of introducing these 2 boolean properties ?
Don't we want to expose that the circuitbreaker can be in a half-open state ?

Being in a 'half open' state is of course an 'internal' thing.  IDK, just wondering what your idea / reasoning is here.

Reply:
Ah, it started as an usability/readability thing. As if we would expose the `_state` itself, you would have to expose the enumeration also and have conditions like these:

```csharp
if (CircuitState.State is CircuitBreakerState.Open)
```

While it was a bit more readable hide the enumeration and the state field and only expose those two start/stop states indeed.

```csharp
if (CircuitState.IsOpen)
```

Reply:
Ok I see and understand.
However, when the CircuitBreaker is in a half-open state, both IsOpen & IsClosed will return false.  How do we cope with that ?

Reply:
Will create `IsHalfOpen`.

Reply:
Done in https://github.com/arcus-azure/arcus.messaging/pull/453/commits/0f1aed451b4444aa8e13f18aba73171cc6558007